packages/application-generic/src/usecases/conditions-filter/conditions-filter.command.ts

```diff
@@ -1,13 +1,18 @@
-import { StepFilter } from '@novu/dal';
 import { IsDefined } from 'class-validator';
-import { EnvironmentWithUserCommand } from '../../commands';
+
+import { StepFilter } from '@novu/dal';
 import { IJob, INotificationTemplateStep } from '@novu/shared';
 
+import { EnvironmentWithUserCommand } from '../../commands';
+import { IFilterVariables } from '../../utils/filter-processing-details';
+
 export class ConditionsFilterCommand extends EnvironmentWithUserCommand {
   @IsDefined()
   filters: StepFilter[];
 
   job?: IJob;
 
   step?: INotificationTemplateStep;
+
+  variables?: IFilterVariables;
```

Comment:
We allow the passing of the data if exists. if not, we will fetch it in the use case if needed.